app/src/main/java/com/noalecohen/dispatcher/view/activity/SearchActivity.kt

```diff
@@ -0,0 +1,105 @@
+package com.noalecohen.dispatcher.view.activity
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.noalecohen.dispatcher.adapter.ArticleAdapter
+import com.noalecohen.dispatcher.databinding.ActivitySearchBinding
+import com.noalecohen.dispatcher.decoration.TopSpacingItemDecoration
+import com.noalecohen.dispatcher.viewmodel.ArticlesViewModel
+import com.noalecohen.dispatcher.viewstate.RequestState
+
+
+class SearchActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySearchBinding
+    private val articlesModel: ArticlesViewModel by viewModels()
+    private var adapter = ArticleAdapter { position -> onListItemClick(position) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySearchBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setBackButton()
+        setCancelButton()
+        subscribeObservers()
+        setRecyclerView()
+        setDoneAction()
+    }
+
+    private fun setCancelButton() {
+        binding.searchInputLayuot.setEndIconOnClickListener {
+            binding.searchInputEditText.text?.clear()
+        }
+    }
+
+    private fun setBackButton() {
+        binding.searchInputLayuot.setStartIconOnClickListener {
+            val intent = Intent(this, MainActivity::class.java)
+            startActivity(intent)
+        }
+    }
+
+    private fun setDoneAction() =
+        binding.searchInputEditText.setOnEditorActionListener { v, actionId, event ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                articlesModel.fetchTopHeadLinesByKeyword(binding.searchInputEditText.text.toString())
```

Comment:
In search, we want to search in everything and not only in the top headlines.
In general in this product if the user filter to only top headlines than the logic you code is correct